packages/app-builder-lib/electron-osx-sign/util.js

```diff
@@ -215,10 +212,7 @@ module.exports.walkAsync = async function (dirPath) {
             await fs.unlink(filePath)
             return
           default:
-            if (path.extname(filePath).indexOf(' ') >= 0) {
-              // Still consider the file as binary if extension seems invalid
-              return getFilePathIfBinaryAsync(filePath)
-            }
+            return getFilePathIfBinarySync(filePath)
```

Comment:
Dropping the extension test will code sign all binary-like files including pictures and icons eventually, we should still include `if (path.extname(filePath).indexOf(' ') >= 0) {` check.

Reply:
Nice catch! I've updated and also consolidated some of the logic into `getFilePathIfBinarySync` for `unpackDetector` to also leverage